apps/gallery/js/ImageEditor.js

```diff
@@ -298,6 +300,14 @@ function editPhoto(n) {
     var imagesize = metadata.width * metadata.height;
     var maxsize = CONFIG_MAX_EDIT_PIXEL_SIZE || CONFIG_MAX_IMAGE_PIXEL_SIZE;
 
+    if (file.type === 'image/jpeg') {
+      LazyLoader.load(['shared/js/media/jpeg-exif.js'], () => {
+        JPEGParser.readExifMetaData(file, (error, metaData) => {
+          editSettings.EXIF = metaData;
```

Comment:
If there is an error for some reason, we don't want to repeat the call to readExifMetaData in the code below, so you should be sure to set editSettings.EXIF in either case. Maybe you just want:

```
editSettings.EXIF = metaData || {};
```